x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/extractor/GeoPointField.java

```diff
@@ -24,9 +24,13 @@ public GeoPointField(String name) {
     @Override
     public Object[] value(SearchHit hit) {
         Object[] value = super.value(hit);
-        if (value.length != 1) {
+        if (value.length == 0) {
+            return new Object[0];
```

Comment:
Theoretically it would save a memory allocation and garbage collection to just `return value;`.  Maybe Hotspot will optimise this in practice.

Same in the other file.